src/payment/data/sagas.js

```diff
@@ -43,6 +48,8 @@ export function* handleFetchBasket() {
 }
 
 export function* handleAddCoupon({ payload }) {
+  if (yield isBasketProcessing()) return;
```

Comment:
Future: We may want to DRY the 3 very similar handlers here since they all have the same structure (by possibly having a helper function that takes in functions as arguments).  Otherwise, it may be prone to future maintenance issues by requiring all handlers to copy-paste calls to `isBasketProcessing`.

Reply:
👍